File: python/parse_json.py
import json
import os
from py2neo import Graph
from py2neo import Node, Relationship

graph = Graph()

fileDir = "/home/larkin/Dropbox/DAH PhD/Neo4J/Metadata/tate/collection/artworks/"
total = 0

mediums = []
subjects = []

for root, dirs, filenames in os.walk(fileDir):
	for f in filenames:
		artwork = json.loads(open(os.path.join(root, f)).read())
		total = total + 1
		 
		#for c in artwork["contributors"]:
		#	print c["id"]

		if artwork["subjectCount"] > 0:
			if "children" in artwork["subjects"]:
				for sl1 in artwork["subjects"]["children"]:
					#print sl1["name"]
					if "children" in sl1:
						for sl2 in sl1["children"]:
							#print "-" + sl2["name"]
							if "children" in sl2:
								for sl3 in sl2["children"]:
									#print "--" + sl3["name"]
									subjects.append(sl3["id"])
		
		node = Node("Artwork", id=artwork["id"], title=artwork["title"], acno=artwork["acno"])		
		graph.create(node)

		for s in subjects:
			subject = graph.merge_one("Subject", "id", s)
			r = Relationship(node, "FEATURES", subject)
			graph.create(r)

		if artwork["medium"]:
			for m in artwork["medium"].split(","):
				for n in m.split(" and "):
					for o in n.split(" on "):
						s = ''.join([i for i in o if not i.isdigit()])
						if s.strip().lower() not in mediums:
							mediums.append(s.strip().lower())
		
		for m in mediums:
			medium = graph.merge_one("Medium", "id", m)
			r = Relationship(node, "MADE_OF", medium)
			graph.create(r)

File: python/parse_json.rb
require 'rubygems'
require 'json'

total = 0

Dir.foreach('/home/larkin/Dropbox/DAH PhD/Neo4J/Metadata/tate/collection/artworks/p/012') do |item|
  	next if item == '.' or item == '..'

	file = open("/home/larkin/Dropbox/DAH PhD/Neo4J/Metadata/tate/collection/artworks/p/012/" + item)
	json = file.read
	total = total + 1
	p total
	parsed = JSON.parse(json)
	p item
	p parsed["title"]

	if parsed["movementCount"] > 0
		if parsed["movements"].kind_of?(Array)
			parsed["movements"].each do |movement|
				p movement["name"]
			end
		end
	end

	if parsed["subjectCount"] > 0
		parsed["subjects"]["children"].each do |subject|
			p subject["name"]
			if subject["children"].kind_of?(Array)
				subject["children"].each do |level2|
					p "-" + level2["name"]
					if level2["children"].kind_of?(Array)
						level2["children"].each do |level3|
							p "--" + level3["name"]
							if level3["children"].kind_of?(Array)
								level3["children"].each do |level4|
									p "---" + level4["name"]
									if level4["children"].kind_of?(Array)
										level4["children"].each do |level5|
											p "---" + level5["name"]
										end
									end
								end
							end
						end
					end
				end
			end
		end
	end
end
